compiler/hash-typecheck/src/ops/reader.rs

```diff
@@ -1,105 +1,102 @@
 //! Contains helpers to read various things stored in [crate::storage] with
 //! ease.
 
+use hash_types::{
+    arguments::ArgsId,
+    mods::ModDefId,
+    nominals::NominalDefId,
+    params::ParamsId,
+    pats::{PatArgsId, PatId},
+    scope::ScopeId,
+    terms::{TermId, TermListId},
+    trts::TrtDefId,
+    Args, ModDef, NominalDef, Params, Pat, PatArgs, Scope, Term, TrtDef,
+};
 use hash_utils::store::{CloneStore, SequenceStore, Store};
 
 use crate::{
     exhaustiveness::{construct::DeconstructedCtor, deconstruct::DeconstructedPat},
     storage::{
-        arguments::ArgsId,
-        deconstructed::{DeconstructedCtorId, DeconstructedPatId},
-        mods::ModDefId,
-        nominals::NominalDefId,
-        params::ParamsId,
-        pats::{PatArgsId, PatId},
-        primitives::{Args, ModDef, NominalDef, Params, Pat, PatArgs, Scope, Term, TrtDef},
-        scope::ScopeId,
-        terms::{TermId, TermListId},
-        trts::TrtDefId,
-        GlobalStorage,
+        exhaustiveness::{DeconstructedCtorId, DeconstructedPatId},
+        StorageRef,
     },
 };
 
 /// Helper to read various primitive constructions (from
 /// [crate::storage::primitives]).
-pub struct PrimitiveReader<'gs> {
-    gs: &'gs GlobalStorage,
+pub struct PrimitiveReader<'tc> {
```

Comment:
This can implement AccessToStorage and then you can use the shorthand methods in AccessToOps!

Reply:
Yeah, before it was just relying on a reference to `gs`, but now due to this refactor, we can now convert it